examples/mag/MAG_parser_v0.2.ipynb

```diff
@@ -0,0 +1,826 @@
+      "There are 23769289 paper citations\n",
+      "There are 1564036 paper-venue pairs\n",
+      "There are 26625613 paper citations\n",
+      "There are 17508917 paper-author pairs\n",
+      "There are 1547869 paper-venue pairs\n",
+      "There are 23454580 paper-fos pairs\n",
+      "There are 24395338 paper-fos pairs\n",
+      "There are 27172944 paper citations\n",
+      "There are 29466019 paper citations\n"
+     ]
+    }
+   ],
+   "source": [
+    "from graphstorm.gconstruct.utils import multiprocessing_data_read\n",
+    "data = multiprocessing_data_read([i for i in range(51)], num_processes=8, user_parser=parse_paper_file)\n",
+    "\n",
+    "num_paper_venue = 0\n",
+    "venue_ids = set()\n",
+    "fos_set = set()\n",
+    "for i in data:\n",
+    "    num_paper_venue += data[i][0]\n",
+    "    venue_ids.update(data[i][1])\n",
+    "    fos_set.update(data[i][2])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "There are 143630713 papers that have venues.\n",
+      "There are 53422 venues with IDs\n",
+      "There are 275021 topic fields\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(f'There are {num_paper_venue} papers that have venues.')\n",
+    "print(f'There are {len(venue_ids)} venues with IDs')\n",
+    "print(f'There are {len(fos_set)} topic fields')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "fos data type: <class 'str'>\n"
+     ]
+    }
+   ],
+   "source": [
+    "df = pd.DataFrame({'id': np.array(list(fos_set))})\n",
+    "table = pa.Table.from_pandas(df)\n",
+    "print('fos data type:', type(list(fos_set)[0]))\n",
+    "pq.write_table(table, 'mag_fos.parquet')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The code below creates venues as labels of paper nodes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 43,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "v_papers = []\n",
+    "venues = []\n",
+    "for i in range(51):\n",
+    "    table = pd.read_parquet(f'mag_paper2venue_{i}.parquet')\n",
+    "    v_papers.append(table['paper'].to_numpy())\n",
+    "    venues.append(table['venue'].to_numpy())\n",
+    "v_papers = np.concatenate(v_papers)\n",
+    "venues = np.concatenate(venues)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 46,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "uniq_venues, v_cnts = np.unique(venues, return_counts=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 61,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "common_venue_map = {venue: i for i, venue in enumerate(uniq_venues[v_cnts > 10000].tolist())}"
```

Comment:
Why v_cnts > 10000?

Reply:
i want to keep the popular venues to reduce the number of classes. if we keep the venues that have at least 10000, there are about 1000 unique venues, that is, there are about 1000 classes.